test/src/test/java/hudson/tasks/MavenTest.java

```diff
@@ -1,404 +0,0 @@
- * 
- * The above copyright notice and this permission notice shall be included in
- * all copies or substantial portions of the Software.
- * 
- * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
- * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
- * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
- * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
- * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
- * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
- * THE SOFTWARE.
- */
-package hudson.tasks;
-
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertFalse;
-import static org.junit.Assert.assertNotEquals;
-import static org.junit.Assert.assertNotNull;
-import static org.junit.Assert.assertNull;
-import static org.junit.Assert.assertTrue;
-
-import com.gargoylesoftware.htmlunit.html.HtmlButton;
-import com.gargoylesoftware.htmlunit.html.HtmlForm;
-import com.gargoylesoftware.htmlunit.html.HtmlPage;
-import hudson.EnvVars;
-import hudson.model.Build;
-import hudson.model.Cause.LegacyCodeCause;
-import hudson.model.FreeStyleBuild;
-import hudson.model.FreeStyleProject;
-import hudson.model.JDK;
-import hudson.model.ParametersAction;
-import hudson.model.ParametersDefinitionProperty;
-import hudson.model.PasswordParameterDefinition;
-import hudson.model.Result;
-import hudson.model.StringParameterDefinition;
-import hudson.model.StringParameterValue;
-import hudson.slaves.EnvironmentVariablesNodeProperty;
-import hudson.tasks.Maven.MavenInstallation;
-import hudson.tasks.Maven.MavenInstaller;
-import hudson.tools.InstallSourceProperty;
-import hudson.tools.ToolProperty;
-import hudson.tools.ToolPropertyDescriptor;
-import hudson.util.DescribableList;
-import java.util.Collections;
-import javax.xml.transform.Source;
-import javax.xml.transform.stream.StreamSource;
-import jenkins.mvn.DefaultGlobalSettingsProvider;
-import jenkins.mvn.DefaultSettingsProvider;
-import jenkins.mvn.FilePathGlobalSettingsProvider;
-import jenkins.mvn.FilePathSettingsProvider;
-import jenkins.mvn.GlobalMavenConfig;
-import org.junit.Rule;
-import org.junit.Test;
-import org.jvnet.hudson.test.ExtractResourceSCM;
-import org.jvnet.hudson.test.Issue;
-import org.jvnet.hudson.test.JenkinsRule;
-import org.jvnet.hudson.test.ToolInstallations;
-import org.kohsuke.stapler.jelly.JellyFacet;
-
-/**
- * @author Kohsuke Kawaguchi
- */
-public class MavenTest {
-
-    @Rule public JenkinsRule j = new JenkinsRule();
-
-    /**
-     * Tests the round-tripping of the configuration.
-     */
-    @Test public void configRoundtrip() throws Exception {
-        j.jenkins.getDescriptorByType(Maven.DescriptorImpl.class).setInstallations(); // reset
-
-        FreeStyleProject p = j.createFreeStyleProject();
-        p.getBuildersList().add(new Maven("a", null, "b.pom", "c=d", "-e", true));
```

Comment:
See https://github.com/jenkinsci/maven-plugin/pull/233#issuecomment-981911743 and #2352.